### ISSUE TITLE
when using sparkNet.fitContinuousLabeledPoint(trainingData),It has a error:merge Feature fails

### ISSUE THREAD

**Issue by zhaoyunyemu** [deeplearning4j/deeplearning4j#9168]
#### Issue Description

when using dl4j spark:
SparkDl4jMultiLayer
sparkNet.fitContinuousLabeledPoint(trainingData),it execution error:
Caused by: java.lang.IllegalStateException: Cannot merge examples: features rank must be in range 2 to 4 inclusive. First example features shape: [4]
	at org.nd4j.linalg.dataset.api.DataSetUtil.mergeFeatures(DataSetUtil.java:211)
#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version： <dl4j.version>1.0.0-beta7</dl4j.version>
* Platform information ：windows 10
* CUDA version, if used： used cpu
* NVIDIA driver version, if in use: used cpu

#### Additional Information
Caused by: java.lang.IllegalStateException: Cannot merge examples: features rank must be in range 2 to 4 inclusive. First example features shape: [4]
	at org.nd4j.linalg.dataset.api.DataSetUtil.mergeFeatures(DataSetUtil.java:211)
	at org.nd4j.linalg.dataset.DataSet.merge(DataSet.java:199)
	at org.deeplearning4j.spark.data.BatchAndExportDataSetsFunction.processList(BatchAndExportDataSetsFunction.java:140)
	at org.deeplearning4j.spark.data.BatchAndExportDataSetsFunction.call(BatchAndExportDataSetsFunction.java:97)
	at org.deeplearning4j.spark.data.BatchAndExportDataSetsFunction.call(BatchAndExportDataSetsFunction.java:47)
	at org.apache.spark.api.java.JavaRDDLike.$anonfun$mapPartitionsWithIndex$1(JavaRDDLike.scala:102)
	at org.apache.spark.api.java.JavaRDDLike.$anonfun$mapPartitionsWithIndex$1$adapted(JavaRDDLike.scala:102)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsWithIndex$2(RDD.scala:889)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsWithIndex$2$adapted(RDD.scala:889)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:127)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**Comment by agibsonccc** [comment]
@zhaoyunyemu  could you give me code to reproduce this?

**Comment by ABI700badu** [comment]
> @zhaoyunyemu could you give me code to reproduce this?

`package com.app.stock.ocr;

import org.deeplearning4j.datasets.iterator.impl.ListDataSetIterator;
import org.deeplearning4j.nn.conf.MultiLayerConfiguration;
import org.deeplearning4j.nn.conf.NeuralNetConfiguration;
import org.deeplearning4j.nn.conf.inputs.InputType;
import org.deeplearning4j.nn.conf.layers.DenseLayer;
import org.deeplearning4j.nn.conf.layers.OutputLayer;
import org.deeplearning4j.nn.multilayer.MultiLayerNetwork;
import org.deeplearning4j.optimize.listeners.ScoreIterationListener;
import org.nd4j.linalg.activations.Activation;
import org.nd4j.linalg.api.ndarray.INDArray;
import org.nd4j.linalg.dataset.DataSet;
import org.nd4j.linalg.factory.Nd4j;
import org.nd4j.linalg.learning.config.Adam;
import org.nd4j.linalg.lossfunctions.LossFunctions;

import java.util.ArrayList;
import java.util.List;

public class TEST2 {
    public static void main(String[] args) {
        int numInput = 2;
        int numHidden = 4;
        int numOutput = 1;
        int numEpochs = 10000;

        // 创建神经网络配置
        MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder()
                .seed(123)
                .weightInit(org.deeplearning4j.nn.weights.WeightInit.XAVIER)
                .updater(new Adam(0.1))
                .list()
                .layer(0, new DenseLayer.Builder()
                        .nIn(numInput)
                        .nOut(numHidden)
                        .activation(Activation.RELU)
                        .build())
                .layer(1, new OutputLayer.Builder(LossFunctions.LossFunction.MSE)
                        .nIn(numHidden)
                        .nOut(numOutput)
                        .activation(Activation.IDENTITY)
                        .build())
                .setInputType(InputType.feedForward(numInput)) // 设置输入类型
                .build();

        MultiLayerNetwork model = new MultiLayerNetwork(conf);
        model.init();
        model.setListeners(new ScoreIterationListener(100));

        // 准备训练数据
        List<DataSet> dataSets = new ArrayList<>();
        dataSets.add(new DataSet(Nd4j.create(new double[]{0.0, 0.0}), Nd4j.create(new double[]{0.0},1)));
        dataSets.add(new DataSet(Nd4j.create(new double[]{0.0, 1.0}), Nd4j.create(new double[]{0.0},1)));
        dataSets.add(new DataSet(Nd4j.create(new double[]{1.0, 0.0}), Nd4j.create(new double[]{0.0},1)));
        dataSets.add(new DataSet(Nd4j.create(new double[]{1.0, 1.0}), Nd4j.create(new double[]{1.0},1)));

        ListDataSetIterator<DataSet> iterator = new ListDataSetIterator<>(dataSets, 4);

        // 训练神经网络
        for (int i = 0; i < numEpochs; i++) {
          //  iterator.reset();
            model.fit(iterator);
        }

        // 测试神经网络
        INDArray testInput = Nd4j.create(new double[][]{{0.5, 2.0}});
        INDArray predictedOutput = model.output(testInput);
        System.out.println("输入值 0.5 和 2.0 的乘法预测结果是: " + predictedOutput.getDouble(0));
    }
}
`

you can try this to  reproduce the bug

**Comment by agibsonccc** [comment]
The spark module is deprecated as of M3 (the next release) due to infrequent usage. I am closing this issue. If your team needs spark support please consider sponsoring a support contract found here:  https://www.konduit.ai/dl4j-consulting
